### ISSUE TITLE
long returned as string

### ISSUE THREAD

**Issue by burner** [denizzzka/dpq2#142]
For some reason this program returns the DevName.dev value as string, but the typing says Int8.

Not sure what is going on?

```D
//import std;
import std.array;
import std.datetime;
import std.algorithm.iteration : map;
import std.algorithm.sorting : sort;
import std.algorithm.searching : canFind;
import std.exception : assumeUnique;
import std.format : format;
import std.string : split;
import std.file : chdir;
import std.typecons : Nullable, nullable;
import std.stdio;
import std.traits;

import dpq2;
import dpq2.conv.to_d_types;

//import librariesparser;
//import gitwalker;
//import model;
//import dbaccess;
//import db;

struct Options {
	string dbHost = "localhost";
	ushort dbPort = 5432;
	string dbUser = "user";
	string dbPass = "weak";
}

Options opts;

Connection getConnection() {
	return new Connection(format!"postgresql://%s:%s@%s:%d"(
		opts.dbUser, opts.dbPass, opts.dbHost, opts.dbPort));
}

T conv(T)(const(Row) row) {
	import std.conv : to;
	T ret;

	foreach(mem; FieldNameTuple!T) {
		if(row.columnExists(mem)) {
			alias MemT = typeof(__traits(getMember, T, mem));
			long idx = row.columnNum(mem);
			Value val = row[idx];
			writeln(T.stringof, " ", mem, " ", MemT.stringof, " ", idx, " ",
					val.oidType(), " ", row.columnName(idx), " ",
					is(MemT == DateTime) ? "DT" : val.toString(), " ",
					val.data, " ");
			__traits(getMember, ret, mem) = val.as!MemT();
		}
	}

	return ret;
}

struct DevName {
	long dev;
	string name;
}

Nullable!T selectOne(T,string column,V)(V value, ref Connection conn) {
	string s = format!`SELECT * FROM %s WHERE %s = '%s'`(T.stringof, column, value);
	auto answer = conn.exec(s);
	//auto answer = conn.execParams(q);
	return answer.length == 0
		? Nullable!(T).init
		: answer.rangify.map!(a => nullable(a.conv!T())).front;
}

void main() {
	//createSQLTable();
	auto conn = getConnection();
	conn.exec(`
DROP TABLE IF EXISTS DevName;
DROP TABLE IF EXISTS Dev;

CREATE TABLE Dev (
	id BIGSERIAL PRIMARY KEY,
	lastUpdated TIMESTAMP
);

CREATE TABLE DevName (
	dev BIGINT REFERENCES Dev(id),
	name VARCHAR UNIQUE,
	PRIMARY KEY(dev, name)
);

INSERT INTO Dev (lastUpdated) VALUES(now());
INSERT INTO Dev (lastUpdated) VALUES(now());

INSERT INTO DevName (dev,name) VALUES(1, 'Foo');
INSERT INTO DevName (dev,name) VALUES(2, 'Bar');
`);
	Nullable!DevName dn = selectOne!(DevName, "name")("Foo", conn);
	writeln(dn);
}
```

**Comment by denizzzka** [comment]
`exec` always returns textual results (it uses old wire protocol)

`execParams` with `QueryParams` argument returns binary results by default. so, try to use `execParams` in all casses, even if you don't need pass any arguments.

**Comment by denizzzka** [comment]
resolved in #143
